### DISCUSSION TITLE
I'm using linux and how to stop git push from asking for username and password?

### DISCUSSION THREAD

**Discussion by funwkwk** [community/community#128806] [Programming Help]
### Select Topic Area

Question

### Body

Support for password authentication was removed on August 13, 2021

**Comment by davevad93** [comment]
Hi @irfan7o ,

To do git operations authentication is needed, as the error states, support for password authentication is no longer available, so in order to `git clone` or `git push` with `HTTPS` you need to use a `personal access token` as password. Alternatively you can also use `SSH` for which you would need to generate an `ssh key` with its password.

Here's how to create a personal access token:

- [Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

And here how to generate an SSH key:

- [Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

**Reply by funwkwk** [reply]
Thanks @davevad93
